### ISSUE TITLE
数据泄露问题 Data leakage

### ISSUE THREAD

**Issue by biginfor** [twjiang/graphSAGE-pytorch#40]
请教各位大牛：

在dataCenter模块中，将数据集随机划分为训练集，测试集和验证集index。由于graphsage聚合信息的时候会采集邻居节点的特征，假设存在一个”超级节点“，他和大量的节点都存在边，这样聚合邻居节点信息的时候有没有可能采集到验证集节点？有没有数据泄露的风险？求赐教！

In the dataCenter module, the dataset is randomly divided into training set, test set and validation set index. since graphsage will collect the features of the neighbouring nodes when aggregating the information, suppose there is a ‘super node’ which has edges with a large number of nodes, is it possible to collect the validation set nodes when aggregating the information of the neighbouring nodes? Is it possible to collect the validation set nodes when aggregating the information of neighbouring nodes? Is there any risk of data leakage? Please advise me!

**Comment by Charging-up** [comment]
我感觉直推式的方法会泄露，但是归纳式的不会
